Sources/UIAlertControllerView.swift

```diff
@@ -69,22 +72,55 @@ class UIAlertControllerView: UIView {
             let previousElement = (index > 0) ? buttons[index - 1] : header
             button.sizeToFit()
             button.contentHorizontalAlignment = .left
-            button.bounds.size.height += verticalPadding
+            button.frame.size.height += verticalPadding
             button.frame.size.width = max(bounds.size.width, button.frame.size.width)
-            button.frame.origin = CGPoint(x: 0, y: previousElement.frame.maxY)
+            button.frame.origin.y = (previousElement == header)
+                ? previousElement.frame.maxY + verticalPadding
+                : previousElement.frame.maxY
+            button.titleLabelOriginX = horizontalPadding
+        })
+    }
+
+    private func layoutAsAlert() {
+        header.frame.origin = CGPoint(x: horizontalPadding, y: verticalPadding)
+        text.frame.origin.x = horizontalPadding
+        text.frame.width = UIScreen.main.bounds.width * 0.5
+        text.sizeToFit()
+        text.frame.minY = header.frame.maxY + verticalPadding
+        buttons.enumerated().forEach({ (arg) in
+            let (index, button) = arg
+            button.sizeToFit()
+            button.frame.width = max(button.frame.width, 75)
+            button.contentHorizontalAlignment = .center
+            button.frame.minY = text.frame.maxY + verticalPadding
+            if button == buttons.first {
+                button.frame.origin.x = bounds.maxX - button.frame.width - verticalPadding
+            } else {
+                let previousElement = buttons[index - 1]
+                button.frame.origin.x = previousElement.frame.minX - button.frame.width
+            }
         })
     }
 
     override func sizeThatFits(_ size: CGSize) -> CGSize {
         layoutIfNeeded()
-
-        let widestElement = buttons.max { a, b in a.frame.width < b.frame.width }
-        let elementWidth = widestElement?.frame.width ?? header.frame.width
-        let elementHeight = header.frame.height + buttons.reduce(0, { $0 + $1.bounds.height })
-
+        
+        let width: CGFloat
+        let height: CGFloat
+        
+        switch style {
+        case .actionSheet:
+            width = ([text] + buttons).map { $0.frame.width }.max()!
```

Comment:
nit:
Currently `text` is always size `zero` in style `actionSheet`. In case we decide not to render it - it can be excluded from this calculation (perhaps).